include/windows/poll.h

```diff
@@ -17,6 +17,8 @@
 
 static inline int poll(struct pollfd *fds, int nfds, int timeout)
 {
-	return WSAPoll(fds, nfds, timeout);
+	/* if none of fds is passed, returns 0 instead of error
+	 * to allign Windows and Linux poll system call */
+	return (!nfds ? 0 : WSAPoll(fds, nfds, timeout));
```

Comment:
Isn't poll with 0 file descriptors used to block a thread for the timeout time?  Can that be emulated here using sleep or something similar?

Do you know where nfds = 0 is being called from?

Reply:
> Isn't poll with 0 file descriptors used to block a thread for the timeout time? Can that be emulated here using sleep or something similar?

Sure. I hadn't thought of that.

> Do you know where nfds = 0 is being called from?

yes, the sockets provider calls with 0 and it leads to `WSAPoll()` failure (but since we checked `errno` instead of `WSAGetLastError()` - we considered that `poll()` is OK always)